C++/C++.sublime-syntax

```diff
@@ -845,7 +845,7 @@ contexts:
     # All uppercase identifier just before a newline is most likely a macro
     - match: '[[:upper:][:digit:]_]+\s*$'
     # Operator overloading
-    - match: '(?=({{path_lookahead}}\s*::\s*)?{{operator_method_name}}\s*(\(|$))'
+    - match: '(?=({{path_lookahead}}\s*(?:{{generic_lookahead}})?::\s*)?{{operator_method_name}}\s*(\(|$))'
```

Comment:
If the generic look ahead variable is a... Look ahead and matches, the following two colons will likely not match. I only see the difference here and doing more review on mobile is tedious, so please excuse me if I missed something here.

Reply:
I mean, you added Tests for this and they pass, but it just seems weird. Maybe the variable terminology is misleading?

Reply:
I'm not entirely sure what you mean.

Yes, this still doesn't highlight correctly:
```
template <class T> bool A::B<T>::C::operator<(const C& c) { return false; }
```
But this works now:
```
template <class T> bool A::B::C<T>::operator<(const C& c) { return false; }
```